hypervisor/common/schedule.c

```diff
@@ -183,6 +183,10 @@ void schedule(void)
 		release_schedule_lock(pcpu_id, rflag);
 		arch_switch_to(&prev->host_sp, &next->host_sp);
 	} else {
+		/* there is a case, when prev == next, and next's
+		 * status is runnable, then, we should make sure
+		 * the status is running after schedule.*/
```

Comment:
Do you mean prev==next and its status is RUNNABLE? How a RUNNABLE thread execute schedule()?